src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -215,7 +234,16 @@ private void DiscoverTestsOnConcurrentManager(IProxyDiscoveryManager proxyDiscov
                             EqtTrace.Verbose("ParallelProxyDiscoveryManager: Discovery started.");
                         }
 
+                        // First marking source as "partially discovered"
+                        DiscoveredSources[nextSource] = DiscoveryStatus.PartiallyDiscovered;
```

Comment:
AS we discussed, not high pri - but can we start off with NotDiscovered and change to PartiallyDiscovered only on exception?